apps/auth/src/pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,129 @@
+import { UserRole } from '@klicker-uzh/prisma'
+import { PrismaAdapter } from '@next-auth/prisma-adapter'
+import bcrypt from 'bcryptjs'
+import JWT from 'jsonwebtoken'
+import type { NextAuthOptions } from 'next-auth'
+import NextAuth from 'next-auth'
+import { DefaultJWT } from 'next-auth/jwt'
+import { Provider } from 'next-auth/providers/index'
+
+import prisma from 'src/lib/prisma'
+
+export async function decode({
```

Comment:
custom decode/encode functions -> we need these to ensure that JWT contents are not encrypted (which next-auth does by default), as that makes them unreadable from our graphql backend